### ISSUE TITLE
[BUG] Rule default-service-account is triggered even when automountServiceAccountToken is set to false

### ISSUE THREAD

**Issue by smuda** [stackrox/kube-linter#135]
**System info:**
 - OS: MaxOS

**Describe the bug**
The rule default-service-account is triggered even when automountServiceAccountToken is set to false.

**To Reproduce**
1. Create a deployment with the following content:

```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: kube-linter-test
  labels:
    label1: content
spec:
  replicas: 1
  selector:
    matchLabels:
      label1: content
  template:
    metadata:
      labels:
        label1: content
    spec:
      automountServiceAccountToken: false
      containers:
        - name: kube-linter-test-container
          image: "image/image:tag"
```

Run kube-linter:
`kube-linter lint --do-not-auto-add-defaults --include default-service-account  ../kube-linter-test`

**Expected behavior**
Since "automountServiceAccountToken" is set to false, I'd expect to not trigger the rule even though serviceAccount is not set.

It's my understanding that the default service account should not be used and, only if there is need, a dedicated service account should be used. Having no rights at all (=no account) is better for a pod which does not need them.
